webview-ui/src/components/GetSolutionDropdown.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect } from "react";
+import {
+  Dropdown,
+  DropdownList,
+  DropdownGroup,
+  DropdownItem,
+  MenuToggle,
+  MenuToggleAction,
+} from "@patternfly/react-core";
+import { effortLevels, SolutionEffortLevel } from "@editor-extensions/shared";
+import { EnhancedIncident } from "@editor-extensions/shared";
+import { useExtensionState } from "../hooks/useExtensionState";
+import { getSolution } from "../hooks/actions";
+import { EllipsisVIcon, WrenchIcon } from "@patternfly/react-icons";
+
+type GetSolutionDropdownProps = {
+  incidents: EnhancedIncident[];
+  scope: "workspace" | "issue" | "in-between" | "incident";
+};
+
+const GetSolutionDropdown: React.FC<GetSolutionDropdownProps> = ({ incidents }) => {
+  if (!incidents || incidents.length === 0) {
+    console.log("Empty Incidents");
+    return null;
+  }
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [state, dispatch] = useExtensionState();
+  const { isFetchingSolution, isAnalyzing, isStartingServer, isInitializingServer } = state;
+  const onGetSolution = (incidents: EnhancedIncident[], effort: SolutionEffortLevel) => {
+    dispatch(getSolution(incidents, effort));
+  };
+
+  // State to track button disabled status
+  const [isButtonDisabled, setIsButtonDisabled] = useState(true);
+
+  useEffect(() => {
+    setIsButtonDisabled(
+      isFetchingSolution || isAnalyzing || isStartingServer || isInitializingServer,
+    );
+  }, [isFetchingSolution, isAnalyzing, isStartingServer, isInitializingServer]);
```

Comment:
`useMemo` is better here -- or even none since computing the 4 flags isn't compute intensive.

Also useEffect has a potential for a render blink while straight compute and useMemo don't.

```tsx
  const isButtonDisabled = useMemo(() => {
    return isFetchingSolution || isAnalyzing || isStartingServer || isInitializingServer;
  }, [isFetchingSolution, isAnalyzing, isStartingServer, isInitializingServer]);

```

Reply:
My understanding was that with the useExtensionStateContext...I shouldn't need to do this anymore. But I should read more into useMemo versus useEffect.

Reply:
For effect think "I need to go fetch something and then push data into my component" and memo is "I only want to do this calculation when the input values change".